### ISSUE TITLE
x/crypto/acme/autocert

### ISSUE THREAD

**Issue by taralx** [golang/go#18433]
Please answer these questions before submitting your issue. Thanks!

### What version of Go are you using (`go version`)?
1.6

### What operating system and processor architecture are you using (`go env`)?
amd64

### What did you do?
Create a Manager that will not accept the TOS, but that does have a persistent Cache. Then re-use the cache with a Manager that does accept the TOS.

### What did you expect to see?
First Manager fails to register, second Manager succeeds.

### What did you see instead?
Seconds Manager also fails because acme.UpdateReg is never called.
See issue 18379 for the acme side of this.

**Comment by x1ddos** [comment]
I'll be back from vacation on Monday but will try to come with a fix end of
this week.

**Comment by taralx** [comment]
If the only supported configuration is "accept the TOS", then perhaps the function should be replaced with a boolean, and passing "false" should cause an error without attempting registration at all.

**Comment by bradfitz** [comment]
No, you're right... if it's there it should work. But it's only there as a formality, not because we expected anybody to use it.
